lms/djangoapps/grades/models.py

```diff
@@ -336,3 +340,59 @@ def update(
         self.possible_graded = possible_graded
         self.visible_blocks_id = visible_blocks_hash  # pylint: disable=attribute-defined-outside-init
         self.save()
+
+
+class SubsectionGradesEnabledFlag(ConfigurationModel):
+    """
+    Enables subsection grades across the platform.
+    When this feature flag is set to true, individual courses
+    must also have subsection grades enabled for the
+    feature to take effect.
+    """
+
+    @classmethod
+    def feature_enabled(cls, course_id=None):
+        """
+        Looks at the currently active configuration model to determine whether the subsection grades feature is
+        available.
+
+        If the flag is not enabled, the feature is not available.
+        If the flag is enabled and the provided course_id is authorized,
+            the feature is available.
+        If the flag is enabled and course-specific authorization is not required, the feature is available.
+        """
+        if not SubsectionGradesEnabledFlag.is_enabled():
+            return False
+        elif course_id:
+            return CourseAuthorization.subsection_grades_enabled(course_id)
+        return True
+
+    class Meta(object):
+        app_label = "grades"
+
+    def __unicode__(self):
+        current_model = SubsectionGradesEnabledFlag.current()
+        return u"SubsectionGradesFlag: enabled {}".format(
+            current_model.is_enabled()
+        )
+
+
+class CourseAuthorization(ConfigurationModel):
```

Comment:
<img width="515" alt="screen shot 2016-08-26 at 1 50 22 pm" src="https://cloud.githubusercontent.com/assets/12545624/18015077/1200bb36-6b94-11e6-9b9d-50a2182ab5a2.png">

I thought we wanted two models because we currently have a setting on the course and a global setting.